backend/src/core/annotation_task.py

```diff
@@ -188,17 +188,57 @@ def build_url(self):
         return self.aggregate_string_replacements(self.dataset['url'])
 
 
+class VersionAnnotationTask(AnnotationTaskInterface):
+    """An annotation task that gets the version of the annotation"""
+
+    def __init__(self, genomic_unit_json):
+        """initializes the task with the genomic_unit"""
+        AnnotationTaskInterface.__init__(self, genomic_unit_json)
+
+    def annotate(self):
+        """placeholder for annotating a genomic unit with version"""
+        return "not-implemented"
+
+    def versioning_by_method(self, versioning_type):
```

Comment:
I really like how you approached this!!!! 🥳 

Something we can talk about in your upcoming work is how this "if else" statement could be structured as a dictionary like it is done for creating the tasks.  No need to change it for this PR.

https://github.com/uab-cgds-worthey/rosalution/blob/7e2892d3f5f51ea826e227e3f19714e0b3db5d79/backend/src/core/annotation_task.py#L197C1-L202C6

What yours would look like is a dictionary with the mappings, but instead of Constructors being the values, you would pass this class's function calls.

Reply:
Thanks! :D 

Yes, that's a great idea! Will implement in future PR :)